src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -3629,8 +3630,8 @@ private ConstantValue GetConstantValueOrThrow(ConstantHandle handle)
 
                 if (_lazyCaseInsensitiveForwardedTypesToAssemblyIndexMap.TryGetValue(fullName, out var value))
                 {
-                    matchedName = fullName;
-                    return value;
+                    matchedName = value.OriginalName;
+                    return (value.FirstIndex, value.SecondIndex);
```

Comment:
Would it be preferable to just return the tuple including the string here? Seems like it might be cleaner than having a return value with two items in it and an out parameter (and also would prevent an allocation, right?)

Reply:
I guess the normal case of getting it from _lazyForwardedTypesToAssemblyIndexMap would require an allocation then (unless it was changed to include the original string too).

Reply:
No allocation prevented. `ValueTuple` is a value type. I'm going to leave the code as is.

Reply:
Oh, for some reason I was thinking the (...) syntax created a Tuple, not a ValueTuple. Whoops!